lib/media/segment_index.js

```diff
@@ -121,6 +121,22 @@ shaka.media.SegmentIndex.prototype.find = function(time) {
 
 
 /**
+ * Finds a SegmentReference immediately prior to the requested time.
+ *
+ * This function can trigger an update, which may add or remove
+ * SegmentReferences.
+ *
+ * @param {number} time The time in seconds.
+ * @return {shaka.media.SegmentReference} The SegmentReference prior to the
+ *     specified time, or null if no such SegmentReference exists.
+ */
+shaka.media.SegmentIndex.prototype.findBefore = function(time) {
+  var i = shaka.media.SegmentReference.find(this.references, time);
+  return i <= 0 ? null : this.references[i - 1];
```

Comment:
Please write this conditional in the same style as the one above in find().  That will make it easier to compare.